### ISSUE TITLE
[Flycast Bug] Joystick movement triggers D-pad buttons as well.

### ISSUE THREAD

**Issue by ghost** [libretro/Lakka-LibreELEC#899]
basically when playing "record of lodoss war" in flycast, with the stable 2.3 build of lakka for the RPI3B+, i discovered that with a generic USB DxInput gamepad, moving the joystick UP would cause the D-pad to go UP at the same time, and conversely, moving the joystick DOWN would also cause the D-pad to go DOWN as well.

the adverse affects of this is that when you make your character walk upwards in lodoss war, the cursor in the items bar on the right side of the screen, will scroll upwards indefinitely until you stop moving, making it impossible to use any items.

i've tried everything to fix the bug, including reconfiguring the controller or tried using different controller, but nothing works. i also tried one of the nightly builds, i don't remember which version as i have already deleted it due to the problem still being there.

i would be happy to do further testing, please let me know anytime.

PS: i have tested my usb controller in an older (may 22nd 2019) build of reicast for windows, running lodosswar. the bug i mentioned still exists there in reicast. however when i run the same game in redream with my usb controller attached, the bug is not present there. so i'm guessing this is an old bug leftover from reicast which still exists presently in flycast.

I've no luck

**Comment by natinusala** [comment]
Can you maybe try disabling the analog to dpad feature in the input settings?

**Comment by blue-agent** [comment]
i disabled analog to dpad config a lonnng time ago, but it never worked.

as i mentioned earlier, i plugged the same usb controller into my pc and ran reicast, this bug is also present in the windows version of reicast. (build date: may 2019) 

i also tried redream in windows, again using the same usb controller, and the bug is not there. 

so i have ruled out the controller being at fault. the bug is only present in windows version of reicast and raspberry pi version of flycast.

did you encounter this bug on your pi? if not then could you tell me which version of lakka are you running on so i can download that same version of lakka that you have and do further testing?

**Comment by blue-agent** [comment]
i've doubled checked the input settings, here's what it shows:

User 1 device type: Retropad
User 1 analog to digital type: none

is this the one you were referring to? i already setted this but it doesn't seem to fix anything.

**Comment by blue-agent** [comment]
by the way i need to emphasized that i only brought up reicast and redream as proof that my controller is not at fault, as the issue of my controller was brought up and i answered to the best of my ability.

when troubleshooting i always give out many details as i can, unlike others who make low effort posts with vague questions full of typos.

 with that said, again, i only talked about reciast in the context that the bug is there also, to stressed that there might be a possibility that the bug was passed down from reicast to flycast. as for redream, i talked about that in the context of the bug not being there, even though i'm using the SAME usb controller which enountered problems in flycast of Lakka PI3. 

with that said, i hope i have clear things up for everyone. i want you all to know that there is absolutely no malice involved on my end. i'm not advertising reicast or redream or putting it on a high pedestal.  everything i say, i say for the sake of troubleshooing. again, let me stressed that there is no malice involved on my end.

**Comment by blue-agent** [comment]
UPDATE:

i think i figured out what the problem is.

in lakka, the right joystick of my controller is seen as an duplicate D-PAD
also, the left joystick of my controller is seen as duplicate ABXY button set.

i do not know why lakka sees my controller like that, and configured it as such by default, but i am not at fault for this.

i tried the same controller in redream and macross m3 works flawlessly, with the right joystick being seen as a joystick and not a duplicate d-pad. thus i can fly the plane with the joystick and transform into robot mode using the d-pad.

here's the controller i have.

https://www.amazon.com/Controller-Gamepad-Joystick-Joypad-Computer-xBox360/dp/B00IXHP4IK
![51wqVvLorXL _SL1005_](https://user-images.githubusercontent.com/61071798/79676655-1aefe780-819d-11ea-9834-df79db110a41.jpg)

i managed to figure this out all on my own, by myself, because while i'm providing as much useful information as i can, i have received no communication from anyone here, nor did anyone requested me to supply more information.

 to save money, i've had to cut off my home internet contract since april 11 this year, then i bought a 5gb data 3G sim card in order to get online and post this information for you. please understand that i am not lying and that this is a real problem i'm trying to report.

**Comment by OctopusButtons** [comment]
Can confirm issue is present in Flycast 22ad95c, ParaLLEI N64 b804ab1, and Beetle PSX, in RetroArch 1.10.2 on **Mac** (both metal and non-metal builds, running in macOS 10.15.7)

if you bind d-pad left/right/up/down in RetroArch, then whatever buttons you binded for Analog Stick Left/Right will wrongly also activate d-pad left/right input.  The only way to avoid the problem is to unbind D-Pad left/right to nothing.

In Metropolis Street Racer the problem means that all steering inputs activate the HORN HONK (because the game assigns that to left/right dpad).  Or in Ape Escape 1 it means the camera spins when you move your character, though camera movement is supposed to be dpad only.

- I made sure that Analog to Digital Type = None.  Didn’t help
- I unbound and rebound everything, didn’t help
- I delete core and game remap files, didn’t help
- I tested in stand-alone Redream emulator and OpenEmu PS1/N64 and the controls are normal and correct
- I also tested in Daytona USA 2001 to rule-out a game-specific bug in Metropolis Street Racer
- I also tried binding Dpad Left/Right to real-world controlller UP and DOWN dpad to see if Analog left right and Dpad left right were still entangled.  They were.
- Ran OS-level computer app to test controller inputs, everything looks fine
- Ran a ControllerTest-DJ.cdi file in RetroArch Flycast, confirmed that indeed analog stick is causing corresponding dpad inputs. It’s a one-way issue: analog stick does dpad input, but dpad input itself is OK and doesn’t affect analog stick input.
- Also used Goldeneye 64 as a test case, the control setting screen has animated input tests.  Issue is present, using analog stick also makes the dpad indicator become active.

In other words:  Analog stick left/right always secretly automatically activates DPAD left/rigiht/up/down, if Dpad left/right is bound to anything. Only way to fix is to completely unbind Dpad Left/Right so that no button no button (“- -“) is assigned, which means you lose any in-game dpad functions.

I've [opened a RetroArch issue here](https://github.com/libretro/RetroArch/issues/14063#issue-1275817547) because it seems upstream, affecting both Lakka and Mac build of RetroArch.

**Comment by Ntemis** [comment]
You better report it upstream then as its def not a lakka issue

**Comment by OctopusButtons** [comment]
> 

Yes sorry about that, it really seems like the page suddenly changed to Lakka but previously said Flycast or RetroArch or a different area, strangely.  Looking for correct place now.
